### PR TITLE
Add docs command

### PR DESCRIPTION
**Pull request by csweichel**
## Description
This PR adds a `docs` command which helps produce readme files for each combination. To this end, a Go template can be provided which is executed for each combination.
Have a look at the `defaultTemplate` in the `docs.go` for an example.

## Related Issue(s)

Fixes #

## How to test
```bash
go run main.go docs --context example/ --all
```

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: cmd/core/docs.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		tplContent := defaultTemplate
+		if fn, _ := cmd.Flags().GetString("template"); fn != "" {
+			fc, err := io.ReadFile(fn)
+			if err != nil {
+				return fmt.Errorf("error reading template: %v", err)
+			}
+			tplContent = string(fc)
+		}
+
+		tpl, err := template.New("readme").Funcs(sprig.FuncMap()).Parse(tplContent)
```
Comment (Siddhant-K-code) [inline cmd/core/docs.go:89]: pls update the file name

```suggestion
		tpl, err := template.New("README").Funcs(sprig.FuncMap()).Parse(tplContent)
```

File: cmd/core/docs.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		if err != nil {
+			return err
+		}
+
+		for _, cmb := range cs {
+			fn := cmb.Name + ".txt"
```
Comment (Siddhant-K-code) [inline cmd/core/docs.go:95]: save it as a markdown file

```suggestion
			fn := cmb.Name + ".md"
```

File: cmd/core/docs.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
+// Copyright © 2020 Gitpod
```
Comment (utam0k) [inline cmd/core/docs.go:1]:
```suggestion
// Copyright © 2022 Gitpod
```

**Comment by kylos101** [comment]
Hey @csweichel , I added this to the Workspace Project, set the status in In-Progress, and assigned you (so that we don't lose sight of this issue).

cc: @Siddhant-K-code @utam0k

**Comment by kylos101** [comment]
Hey @csweichel @Siddhant-K-code and @axonasif, is this still needed?

**Comment by Siddhant-K-code** [comment]
> Hey @csweichel @Siddhant-K-code and @axonasif, is this still needed?

Yes, this PR was intended to create a markdown file containing the content of our workspace images. Which would help the better documentation for workspace images 🙏🏼

**Comment by kylos101** [comment]
:wave: hey there, removing this from the Team Workspace project (for now). @csweichel not sure if you want to pick this back up, but, it's not currently a priority.

**Pull request closed without merging**